python/ray/train/base_trainer.py

```diff
@@ -122,8 +122,9 @@ def training_loop(self):
     .. code-block:: python
 
         import ray
+        import numpy as np
         train_dataset = ray.data.from_items(
-            [{"x": i, "y": i} for i in range(3)])
+            [{"x": np.float32(i), "y": np.float32(i)} for i in range(3)])
```

Comment:
Could you instead move this cast to the `to_torch()` call, using `label_column_dtype` and `feature_column_dtypes`? https://docs.ray.io/en/master/data/package-ref.html#ray.data.Dataset.to_torch

Reply:
this actually sounds good to me!

Reply:
this example also shows up here: https://github.com/ray-project/ray/blob/master/python/ray/air/examples/custom_trainer.py#L56